src/main/scala/is/hail/annotations/UnsafeRow.scala

```diff
@@ -75,8 +75,8 @@ object UnsafeRow {
   def readString(region: MemoryBuffer, offset: Long): String =
     new String(readBinary(region, offset))
 
-  def readLocus(region: MemoryBuffer, offset: Long): Locus = {
-    val ft = TLocus(GenomeReference.GRCh37).fundamentalType.asInstanceOf[TStruct]
+  def readLocus(region: MemoryBuffer, offset: Long, gr: GRBase): Locus = {
+    val ft = TLocus(gr).fundamentalType.asInstanceOf[TStruct]
```

Comment:
I think we should have `GenotypeReference.variant: TVariant`, etc., for types parameterized by that reference, to avoid allocating new types in situations like this.